Feature/250 ether input

luchux opened the pull request:
Done: 
- Added YUP for validation schemas. 
- Added form validation only to RewardContracts form at the moment. 

ToDo after review: 
- Add for other forms. 
- See if we need extended schemas, and if we can reuse some according to similar forms. 

Actual rules: 
beneficiary: check non empty (we can do ether address verification or things like that if needed)
repChange: check non empty. check that is bigger or equal to 0 (= because the placeholder was set to 0). check that is not higher than 10 (just as an example if we need upper limits) 
ethValue: check non empty. check that is bigger or equal than 0. 
externalToken: check non empty. check that is bigger or equal than 0. 
tokenValue: check non empty. check that is bigger or equal than 0. check is not bigger than 9999. 

in all those fail checks, show the errors as in strings of yup. 
    
Some examples: 
<img width="1230" alt="Screen Shot 2021-10-21 at 4 45 28 AM" src="https://user-images.githubusercontent.com/1475656/138270787-718fa770-1dc6-47ce-9c4d-f48ef5991882.png">
<img width="1273" alt="Screen Shot 2021-10-21 at 4 42 29 AM" src="https://user-images.githubusercontent.com/1475656/138270790-230e51d1-0537-45ed-a62d-24a66a73d88c.png">
<img width="1207" alt="Screen Shot 2021-10-21 at 4 45 15 AM" src="https://user-images.githubusercontent.com/1475656/138270788-e1c5c5a3-4b5a-4351-8203-edbd84bf9472.png">

<img width="1196" alt="Screen Shot 2021-10-21 at 4 45 40 AM" src="https://user-images.githubusercontent.com/1475656/138270783-e9040615-e787-46e6-9873-190c118e153c.png">
<img width="1212" alt="Screen Shot 2021-10-21 at 4 49 08 AM" src="https://user-images.githubusercontent.com/1475656/138271122-6309fd79-6f91-4181-8e1f-3a32fbc67994.png">

luchux committed:
adding validation for ContributorRewards

luchux committed:
removing color

luchux committed:
adding message to token in eth

rossneilson reviewed (commented):
I havent used yup before but it seems like a good way of validating this.
Is it possible to add it also for the other call types for example all the calls supported by the quick wallet schemes etc?

luchux commented:
@rossneilson sure! lets chat about it.

luchux committed:
Merge branch 'develop' of github.com:DXgovernance/dxvote into feature/250-ether-input

luchux committed:
adding form validation in Calls and Schemes

luchux committed:
removing value cast from wei to eth

luchux commented on src/pages/NewProposal.tsx:
```diff
@@ -353,21 +385,15 @@ const NewProposalPage = observer(() => {
-            : call.callType === 'simple'
-            ? library.utils.toWei(call.value).toString()
-            : call.value;
+          return callToController ? '0' : call.value;
```
@AugustoL @rossneilson since we are using eth instead of wei, this should not cast, right?

luchux commented on src/pages/NewProposal.tsx:
```diff
@@ -353,21 +385,15 @@ const NewProposalPage = observer(() => {
-              call.callType === 'simple'
-                ? library.utils.toWei(call.value).toString()
-                : call.value
+              call.value
```
@AugustoL @rossneilson since we are using eth instead of wei, this should not cast, right?

rossneilson replied:
I think this should be getting transformed to wei before we encode it right?
Ether is the human readable version but contracts want wei and so it needs the toWei function

luchux replied:
ok!

luchux commented:
@rossneilson you can see how i added more schemes, and added minimum to some fields in calls. 

If we want all the fields marked we can do this as well. 
I will give a retouch to the styles of the forms still (error color border in input as well), and disable push to ifps  butotn if there are errors. 

So if you wanna discuss the rules for addresses, and values and messages, would be epic ;) Also 0 values, placeholder are 0, so can we send 0 value? same for SchemeRegister...

rossneilson commented:
I may have missed something but I think these are still only being applied to the old daostack calls (Majority of schemes in mainnet and xdai). However since we are moving towards the new architecture and schemas it would be nice for this to work there also.
See the below screenshot for an example of where it would be nice to apply the validation to:
<img width="1321" alt="スクリーンショット 2021-10-28 13 25 31" src="https://user-images.githubusercontent.com/39137239/139254765-d512c014-a517-4eb8-b1b8-d7886988bcf3.png">

luchux committed:
merge devleop

luchux committed:
merging develop
